### PR TITLE
Add Completion Method

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Add methods to determine if previous task status has been completed
* Add additional specs for new public methods
* Add specs for membership methods
* Remove unnecessary method and specs  `available_for_learning`

[#114008595]

### CODE REVIEW

File: app/models/task_status.rb [hunk 1]
```diff
@@ -67,6 +56,22 @@ class TaskStatus < ActiveRecord::Base
...
+    available_for_learning_on <= Date.today
+  end
+
+  def completed?
+    completed_at.present?
+  end
+
+  def previous_completed?
+    return true if is_first?
+    self
```
Comment (eschlange) [inline app/models/task_status.rb]: What is the purpose of invoking this method off of the model's class?

Comment (michaelwehrley) [inline app/models/task_status.rb]: Sorry, but I am not sure I understand the question.  I am running a query here and I need to do that off the `TaskStatus`.  Are you saying to instead write:
```ruby
  def previous_completed?
    return true if is_first?
    TaskStatus
      .find_by_id(membership_task_status_ids[index_for_membership - 1])
      .completed?
  end
```

Comment (eschlange) [inline app/models/task_status.rb]: Ah, understood. I haven't seen that notation used for this purpose.

**Comment by eschlange** [comment]
Looks good, nice refactoring and specs as well.

**Pull request merged**